### PR TITLE
DTSPO-1452 - Divorce-COS SendGrid Migration

### PR DESCRIPTION
**Pull request by chrisorisawayi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-1452

### Change description ###
Added sendgrid mail config to Divorce-COS manifest.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: k8s/aat/common/divorce/div-cos.yaml [hunk 1]
```diff
@@ -41,12 +41,22 @@ spec:
...
+        DATAEXTRACTION_STATUS_DA_EMAILFROM: noreply@mail-divorce-nonprod.platform.hmcts.net
         DATAEXTRACTION_STATUS_DA_EMAILTO: da_data_extraction@sharklasers.com
         DATAEXTRACTION_STATUS_DN_EMAILTO: dn_data_extraction@sharklasers.com
         DATAEXTRACTION_STATUS_AOS_EMAILTO: aos_data_extraction@sharklasers.com
         SCHEDULER_ENABLED: true
         SCHEDULER_SEND_UPDATED_CASES_TO_ROBOTICS_CRON: "0 0 4 ? * * *"
         SCHEDULER_MAKE_ELIGIBLE_CASES_AOS_OVERDUE_ENABLED: true
+        SPRING_MAIL_HOST: smtp.sendgrid.net
+        SPRING_MAIL_PORT: 587
+        SPRING_MAIL_PROPERTIES_MAIL_SMTP_SSL_TRUST: smtp.sendgrid.net
```
Comment (timja) [inline k8s/aat/common/divorce/div-cos.yaml:53]:
```suggestion
        SPRING_MAIL_PROPERTIES_MAIL_SMTP_SSL_TRUST: smtp.sendgrid.net
        SPRING_MAIL_USERNAME: apikey
```

File: k8s/aat/common/divorce/div-cos.yaml [hunk 2]
```diff
@@ -41,12 +41,22 @@ spec:
...
+      keyVaults:
+        "div":
+          resourceGroup: div
+          secrets:
+          - spring-mail-username
```
Comment (timja) [inline k8s/aat/common/divorce/div-cos.yaml:58]: you need to include all secrets from the base chart and then append spring mail password

https://github.com/hmcts/div-case-orchestration-service/blob/master/charts/div-cos/values.yaml#L34-L48

File: k8s/aat/common/divorce/div-cos.yaml [hunk 3]
```diff
@@ -41,12 +41,23 @@ spec:
         SPRING_MAIL_PROPERTIES_MAIL_SMTP_STARTTLS_ENABLE: false
         SCHEDULER_SEND_UPDATED_CASES_TO_ROBOTICS_ENABLED: true
         BULKACTION_MINCASES: 5
+        DATAEXTRACTION_STATUS_DA_EMAILFROM: noreply@mail-divorce-nonprod.platform.hmcts.net
```
Comment (timja) [inline k8s/aat/common/divorce/div-cos.yaml:44]:
```suggestion
        DATAEXTRACTION_EMAILFROM: noreply@mail-divorce-nonprod.platform.hmcts.net
```

File: k8s/aat/common/divorce/div-cos.yaml (hunk 2 above)
Comment (chrisorisawayi) [reply k8s/aat/common/divorce/div-cos.yaml]: added

**Review by timja — APPROVED** [review]

**Review by jvictorcf — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

**Pull request merged**